phoenix-core/src/it/java/org/apache/phoenix/end2end/UpgradeIT.java

```diff
@@ -697,21 +698,33 @@ public void testLastDDLTimestampBootstrap() throws Exception {
             conn.createStatement().execute(
                 "CREATE VIEW " + fullViewName + " AS SELECT * FROM " + fullTableName);
 
+            conn.createStatement().execute("CREATE INDEX " + indexName + " ON " + fullTableName + " (KV1) ASYNC");
+
             //Now we null out any existing last ddl timestamps
             nullDDLTimestamps(conn);
 
             //now get the row timestamps for each header row
             long tableTS = getRowTimestampForMetadata(conn, schemaName, tableName,
                 PTableType.TABLE);
             long viewTS = getRowTimestampForMetadata(conn, schemaName, viewName, PTableType.VIEW);
+            long indexTS = getRowTimestampForMetadata(conn, schemaName, indexName, PTableType.INDEX);
 
-            UpgradeUtil.bootstrapLastDDLTimestamp(conn.unwrap(PhoenixConnection.class));
+            // bootstrap last ddl timestamp for tables and views
+            UpgradeUtil.bootstrapLastDDLTimestampForTablesAndViews(conn.unwrap(PhoenixConnection.class));
             long actualTableTS = getLastTimestampForMetadata(conn, schemaName, tableName,
                 PTableType.TABLE);
             long actualViewTS = getLastTimestampForMetadata(conn, schemaName, viewName,
                 PTableType.VIEW);
+            long actualIndexTS = getLastTimestampForMetadata(conn, schemaName, indexName,
+                    PTableType.INDEX);
             assertEquals(tableTS, actualTableTS);
             assertEquals(viewTS, actualViewTS);
+            assertEquals(0L, actualIndexTS);
+
+            // bootstrap last ddl timestamp for indexes
+            UpgradeUtil.bootstrapLastDDLTimestampForIndexes(conn.unwrap(PhoenixConnection.class));
+            actualIndexTS = getLastTimestampForMetadata(conn, schemaName, indexName, PTableType.INDEX);
+            assertEquals(indexTS, actualIndexTS);
```

Comment:
Maybe I am missing something. 
`indexTS` is the timestamp after we set LAST_DDL_TIMESTAMP to null via `nullDDLTimestamps` method. So it should be NULL.
`actualIndexTS` is the timestamp after we call `UpgradeUtil.bootstrapLastDDLTimestampForIndexes`. So it should the LAST_ROW_TIMESTAMP() for this index.
So how are they supposed to be equal?

Reply:
@shahrs87 `indexTS` is the PHOENIX_ROW_TIMESTAMP() of the header row of the index in syscat (it is the timestamp of the empty column) and indicates the last time the row was modified. After we call `UpgradeUtil.bootstrapLastDDLTimestampForIndexes`, this same value for the row is used to populate the `LAST_DDL_TIMESTAMP` column which is captured in `actualIndexTS`.

Reply:
Oh my bad, I thought indexTS is calculated from `getLastTimestampForMetadata` method.